### PR TITLE
HOCS 3417 Allocate a case

### PR DESCRIPTION
**Pull request by LaurieBennett**
Updates the allocation stage, removing the Request Type Selection screen, removing the Allocation Case Note entry field and adding in a confirmation screen displaying the team name and case number.

### CODE REVIEW

**Comment by gary-newbolt** [comment]
In `src/main/resources/processes/FOI_ALLOCATION.bpmn` have you left the backwards flow in for a reason? I can't see the purpose or activation of this backwards flow now.

**Comment by gary-newbolt** [comment]
There also seems to be a lot of changes on the `FOI.bpmn` for what looks like 0 actual changes, it's hard to tell though. Did you end up making any changes to that file? If not can you revert the changes to it so it's excluded from the merge request please?

File: src/test/java/uk/gov/digital/ho/hocs/workflow/processes/FOI_ALLOCATION.java [hunk 1]
```diff
@@ -77,25 +77,24 @@ public void happyPath() {
...
         //then
-        verify(processScenario, times(1)).hasCompleted(CHOOSE_REQUEST_TYPE);
         verify(processScenario, times(1)).hasCompleted(CHOOSE_FOI_HUB);
         verify(bpmnService).updateAllocationNoteWithDetails(eq(CASE_UUID), eq(STAGE_UUID), eq(ALLOCATION_MESSAGE),
                 eq("ALLOCATE"), eq(ACCEPTANCE_TEAM_UUID), eq(STAGE_UUID));
+        verify(processScenario, times(1)).hasCompleted(CONFIRMATION_SCREEN);
     }
 
     @Test
     public void happyPathWithBack() {
```
Comment (gary-newbolt) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/processes/FOI_ALLOCATION.java:87]: If you take out the back flow, you will need to remove this test as well. There may be others you'll need to check.

Comment (LaurieBennett) [reply src/test/java/uk/gov/digital/ho/hocs/workflow/processes/FOI_ALLOCATION.java]: Done

**Review by gary-newbolt — COMMENTED** [review]
lgtm... just awaiting front end review changes.

**Review by gary-newbolt — APPROVED** [review]
lgtm

**Pull request merged**